### PR TITLE
celestia-da: add client context

### PR DESCRIPTION
**Pull request by tuxcanfly**
## Overview

This PR updates `celestia-da` to support client contexts for DA interface methods.

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

Depends on:

- https://github.com/rollkit/go-da/pull/28
- https://github.com/rollkit/local-celestia-devnet/pull/84

## Summary by CodeRabbit

- **Refactor**
  - Enhanced various data access methods to include context for improved operation management.
  
- **Documentation**
  - Updated method signatures to reflect the inclusion of context parameters.

### CODE REVIEW

File: celestia_test.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ func (t *TestSuite) SetupSuite() {
 	}
 
 	// pulls an image, creates a container based on it and runs it
-	resource, err := pool.Run("ghcr.io/rollkit/local-celestia-devnet", "4ecd750", []string{})
+	resource, err := pool.Run("ghcr.io/rollkit/local-celestia-devnet", "394c0b8", []string{})
```
Comment (ramin) [inline celestia_test.go:48]: it'd be neat to move this to a const at the top of the _test.go file so its easy to source/find and suggests that its being used in the test

```
const(
  localDevnetImageVersion = "394c0b8"
)
```

or something?

**Review by ramin — COMMENTED** [review]
I think the rabbit's comments about using the passed in ctx vs anything stored on c.ctx is the preferred approach here

File: celestia_test.go (hunk 1 above)
Comment (tuxcanfly) [reply celestia_test.go:48]: Sounds good 👍🏽

File: go.mod [hunk 2]
```diff
@@ -349,3 +349,5 @@ replace (
 	github.com/filecoin-project/dagstore => github.com/celestiaorg/dagstore v0.0.0-20230824094345-537c012aa403
 	github.com/tendermint/tendermint => github.com/celestiaorg/celestia-core v1.29.0-tm-v0.34.29
 )
+
+replace github.com/rollkit/go-da => github.com/rollkit/go-da v0.1.1-0.20240111224712-bff650eba569
```
Comment (tuxcanfly) [inline go.mod:353]: needs to removed on new go-da release

**Review by Manav-Aggarwal — APPROVED** [review]

**Review by S1nus — APPROVED** [review]

**Review by ramin — APPROVED** [review]

**Review by Manav-Aggarwal — APPROVED** [review]

**Pull request merged**